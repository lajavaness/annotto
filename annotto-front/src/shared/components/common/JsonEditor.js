import { UnControlled as CodeMirror } from 'react-codemirror2'
import React, { useState } from 'react'
import jsonlint from 'jsonlint-mod'

import 'codemirror/addon/fold/brace-fold.js'
import 'codemirror/addon/fold/comment-fold.js'
import 'codemirror/addon/fold/foldcode.js'
import 'codemirror/addon/fold/foldgutter.js'
import 'codemirror/addon/lint/json-lint.js'
import 'codemirror/addon/lint/lint.js'
import 'codemirror/mode/javascript/javascript'

import 'codemirror/addon/fold/foldgutter.css'
import 'codemirror/addon/lint/lint.css'
import 'codemirror/lib/codemirror.css'
import 'codemirror/theme/neat.css'

window.jsonlint = jsonlint

const JsonEditor = ({ ...props }) => {
	const [mounted, setMounted] = useState(false)

	const _onEditorDidMount = () => setMounted(true)

	return (
		<CodeMirror
			{...props}
			editorDidMount={_onEditorDidMount}
			autoScroll={false}
			autoCursor={false}
			options={{
				mode: 'application/json',
				theme: 'neat',
				lineNumbers: true,
				smartIndent: true,
				lint: mounted,
				tabSize: 2,
				foldGutter: true,
				selfContain: true,
				gutters: ['CodeMirror-linenumbers', 'CodeMirror-foldgutter', 'CodeMirror-lint-markers'],
			}}
		/>
	)
}

export default JsonEditor
