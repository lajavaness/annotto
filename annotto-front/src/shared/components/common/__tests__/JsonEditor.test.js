import { render } from '@testing-library/react'
import React from 'react'

import JsonEditor from 'shared/components/common/JsonEditor'

const defaultProps = {}
const getInstance = (props = {}) => (
	<div>
		<JsonEditor {...defaultProps} {...props} />
	</div>
)

describe('JsonEditor', () => {
	document.createRange = () => {
		const range = new Range()

		range.getBoundingClientRect = jest.fn()

		range.getClientRects = () => {
			return {
				item: () => null,
				length: 0,
				[Symbol.iterator]: jest.fn(),
			}
		}

		range.startContainer.getBoundingClientRect = jest.fn()

		return range
	}

	it('matches snapshot', () => {
		const { asFragment } = render(getInstance())
		expect(asFragment()).toMatchSnapshot()
	})
})
