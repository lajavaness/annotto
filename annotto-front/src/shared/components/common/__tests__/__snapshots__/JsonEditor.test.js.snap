// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JsonEditor matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="react-codemirror2"
    >
      <div
        class="CodeMirror cm-s-neat"
        translate="no"
      >
        <div
          style="overflow: hidden; position: relative; width: 3px; height: 0px;"
        >
          <textarea
            autocapitalize="off"
            autocorrect="off"
            spellcheck="false"
            style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"
            tabindex="0"
          />
        </div>
        <div
          class="CodeMirror-vscrollbar"
          cm-not-content="true"
          tabindex="-1"
        >
          <div
            style="min-width: 1px;"
          />
        </div>
        <div
          class="CodeMirror-hscrollbar"
          cm-not-content="true"
          tabindex="-1"
        >
          <div
            style="height: 100%; min-height: 1px;"
          />
        </div>
        <div
          class="CodeMirror-scrollbar-filler"
          cm-not-content="true"
        />
        <div
          class="CodeMirror-gutter-filler"
          cm-not-content="true"
        />
        <div
          class="CodeMirror-scroll"
          tabindex="-1"
        >
          <div
            class="CodeMirror-sizer"
            style="margin-left: 0px; min-width: 3px;"
          >
            <div
              style="position: relative;"
            >
              <div
                class="CodeMirror-lines"
                role="presentation"
              >
                <div
                  role="presentation"
                  style="position: relative; outline: none;"
                >
                  <div
                    class="CodeMirror-measure"
                  >
                    <span>
                      <span>
                        ​
                      </span>
                      x
                    </span>
                  </div>
                  <div
                    class="CodeMirror-measure"
                  >
                    <pre
                      class="CodeMirror-line"
                      role="presentation"
                    >
                      <span
                        role="presentation"
                        style="padding-right: .1px;"
                      >
                        <span
                          cm-text=""
                          style="display: inline-block; width: 1px; margin-right: -1px;"
                        >
                           
                        </span>
                      </span>
                    </pre>
                  </div>
                  <div
                    style="position: relative; z-index: 1;"
                  />
                  <div
                    class="CodeMirror-cursors"
                  />
                  <div
                    class="CodeMirror-code"
                    role="presentation"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            style="position: absolute; height: 50px; width: 1px;"
          />
          <div
            class="CodeMirror-gutters"
            style="left: 0px;"
          >
            <div
              class="CodeMirror-gutter CodeMirror-linenumbers"
              style="width: 1px;"
            />
            <div
              class="CodeMirror-gutter CodeMirror-foldgutter"
            />
            <div
              class="CodeMirror-gutter CodeMirror-lint-markers"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
